code/game/objects/structures/stool_bed_chair_nest/stools.dm

```diff
@@ -74,3 +73,15 @@
 		T.Weaken(5)
 		return
 	..()
+
+/obj/item/weapon/stool/dropped(mob/user) //Make sure we aren't leaving un-deconstructible stools.
+	..()
+	if(!user.in_throw_mode) //If this stool isn't getting thrown, turn it back into a structure.
+		origin.forceMove(user.loc)
+		qdel(src)
+
+/obj/item/weapon/stool/throw_impact(atom/hit_atom) //Thrown but uncaught stools will automatically turn into the structure version as soon as the throw is over.
+	..()
+	if(!iscarbon(loc)) //If it wasn't caught by someone, put the structure version down.
+		new origin.type(get_turf(src))

```

Comment:
Put the move out into a `spawn()` then to give the throw processing time to complete before it gets moved out (put a comment on the spawn letting people know why it's there)